### ISSUE TITLE
unresolved vars for potemkin-imported vars when linting single file

### ISSUE THREAD

**Issue by ericdallo** [clj-kondo/clj-kondo#1167]
**version**
`2021.01.21-20210207.143254-18`

**platform**

JVM

**editor**

Confirmed the issue with Calva, clojure-lsp and JVM only via terminal.

**problem**

when clj-kondo lint a single file it reports `unresolved-var` even if the whole classpath was previously linted and cached it.

**repro**

First, we add a new unknown function [here](https://github.com/clj-commons/rewrite-clj/blob/v1/test/rewrite_clj/zip_test.cljc#L7) to test the unresolved-var: `(z/unknown)`

Then, we lint the whole classpath: 
```bash
$ clj -Sdeps '{:deps {clj-kondo {:mvn/version "2021.01.21-20210207.143254-18"}}}' -m clj-kondo.main --lint src test

test/rewrite_clj/zip_test.cljc:10:40: warning: Unresolved var: z/unknown
linting took 4882ms, errors: 0, warnings: 1
```
But when lining the single file I get:
```bash
$ cat test/rewrite_clj/zip_test.cljc | clj -Sdeps '{:deps {clj-kondo {:mvn/version "2021.01.21-20210207.143254-18"}}}' -m clj-kondo.main --lint -

<stdin>:10:28: warning: Unresolved var: z/of-string
<stdin>:10:40: warning: Unresolved var: z/unknown
<stdin>:26:26: warning: Unresolved var: z/find-tag-by-pos
<stdin>:27:26: warning: Unresolved var: z/replace
<stdin>:28:26: warning: Unresolved var: z/append-child
<stdin>:29:25: warning: Unresolved var: z/down
<stdin>:30:25: warning: Unresolved var: z/remove
<stdin>:31:25: warning: Unresolved var: z/root-string
<stdin>:47:16: warning: Unresolved var: z/rightmost
<stdin>:48:16: warning: Unresolved var: z/child-sexprs
<stdin>:57:16: warning: Unresolved var: z/up
<stdin>:68:16: warning: Unresolved var: z/right
<stdin>:82:17: warning: Unresolved var: z/insert-child
<stdin>:92:28: warning: Unresolved var: n/map-qualifier-node
<stdin>:106:36: warning: Unresolved var: z/node
<stdin>:106:43: warning: Unresolved var: n/map-context-clear
<stdin>:116:18: warning: Unresolved var: z/sexpr
<stdin>:136:18: warning: Unresolved var: z/reapply-context
linting took 231ms, errors: 0, warnings: 18
```

**config**

clj-kondo should find [the project config](https://github.com/clj-commons/rewrite-clj/blob/v1/.clj-kondo/config.edn) automatically

**expected behavior**

On single lint, warn only about the `z/unknown` `unresolved-var`

**Comment by borkdude** [comment]
The issue is not that the config isn't found, the issue is that the namespace data from the imported namespace isn't read when linting a single file.

**Comment by ericdallo** [comment]
For users with this issue, there is a temporary workaround to disable the `unresolved-var` linter for the potemkin imported namespace:
`.clj-kondo/config.edn`
```clojure
{:linters {:unresolved-var {:exclude [your-ns]}}}
```

**Comment by borkdude** [comment]
The situation should be improved at least with commit https://github.com/clj-kondo/clj-kondo/commit/65b369754735860347a6c517e58a4db1f58bffa8. Maybe try it.

**Comment by borkdude** [comment]
cc @lread

**Comment by ericdallo** [comment]
Tested on @lread rewrite-clj-repro and it worked with clojure-lsp :D will add to next clojure-lsp release, thank you @borkdude !
